File: plugins/dashboard-quality/src/types/nonconformity.ts
import type { PaginationParams } from "./pagination";

export type NonconformityType = "internal" | "external" | "all";

export type ListNonconformitiesParams = PaginationParams & {
  type?: NonconformityType;
  branch?: string;
  date_start?: string;
  date_end?: string;
  status?: string;
  item_code?: string;
  description?: string;
};

export type Nonconformity = {
  branch: string;
  code: string;
  code_display: string | null;
  revision: string;
  type_code: string;
  type_label: string | null;
  status_code: string | null;
  status_label: string | null;
  description: string | null;
  detailed_description: string | null;
  item_code: string | null;
  op_code: string | null;
  registered_date: string | null;
  occurrence_date: string | null;
  priority_code: string | null;
  priority_label: string | null;
  origin_department: string | null;
  destination_department: string | null;
  customer_code: string | null;
  customer_store: string | null;
  customer_name: string | null;
  supplier_code: string | null;
  supplier_store: string | null;
  produced_quantity: number | null;
  returned_quantity: number | null;
};

export type NonconformitySeriesPoint = {
  periodo: string;
  sort_key: string;
  date_start: string;
  date_end: string;
  value: number;
  registros: number;
};

export type NonconformitySeriesResponse = {
  type: NonconformityType;
  granularity: string;
  truncated: boolean;
  points: NonconformitySeriesPoint[];
};

File: api-delpi/app/domain/entities/nonconformity/nonconformity.py
# app/domain/entities/nonconformity/nonconformity.py

from dataclasses import dataclass
from typing import Optional


@dataclass
class Nonconformity:
    branch: str
    code: str
    revision: str
    type_code: str
    code_display: Optional[str] = None
    type_label: Optional[str] = None
    status_code: Optional[str] = None
    status_label: Optional[str] = None
    description: Optional[str] = None
    detailed_description: Optional[str] = None
    item_code: Optional[str] = None
    op_code: Optional[str] = None
    registered_date: Optional[str] = None
    occurrence_date: Optional[str] = None
    priority_code: Optional[str] = None
    priority_label: Optional[str] = None
    origin_department: Optional[str] = None
    destination_department: Optional[str] = None
    customer_code: Optional[str] = None
    customer_store: Optional[str] = None
    customer_name: Optional[str] = None
    supplier_code: Optional[str] = None
    supplier_store: Optional[str] = None
    produced_quantity: Optional[float] = None
    returned_quantity: Optional[float] = None